test/jdk/java/util/concurrent/Executors/UnreferencedExecutor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8308235
+ * @summary Unreference ExecutorService objects returned by the Executors without shutdown
+ *    and termination, this should not leak memory
+ * @run main/othervm -Xmx32m UnreferencedExecutor
+ */
+
+import java.util.concurrent.Executors;
+
+public class UnreferencedExecutor {
+    public static void main(String[] args) throws Exception {
+        int ncores = Runtime.getRuntime().availableProcessors();
+        long start = System.currentTimeMillis();
+        while (System.currentTimeMillis() - start < 5000) {
```

Comment:
@AlanBateman Might want to have this based on `nanoTime()` such that changes to wall-clock doesn't impact the length of the run of the test.

Reply:
Okay, that would be better.